### ISSUE TITLE
Matcher for "should be ignoring given field names"

### ISSUE THREAD

**Issue by AlexBroadbent** [kotest/kotest#711]
Can we get a matcher for checking equality, without checking certain fields of the object. 

Such as AssertJ's [isEqualToIgnoringGivenFields](https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractObjectAssert.html#isEqualToIgnoringGivenFields(java.lang.Object,%20java.lang.String...)) method.

This is useful when I am trying to do a `shouldBe` check on two objects which have a `createdOn` date/time field. It would be nice to be able to say something like `obj1.shouldBeIgnoringFields(obj2, "createdOn")`.

**Comment by sksamuel** [comment]
`obj.shouldBeEqualUsingFields(obj2, "name", "age", "job")`
`obj.shouldBeEqualIgnoringFields(obj2, "location")`

**Comment by sksamuel** [comment]
I think we can add this to 3.3.2

**Comment by LeoColman** [comment]
I don't think we should use strings to field names

**Comment by LeoColman** [comment]
That works for private fields tho, but I think we could do something like

```
obj.shoudBeEqualUsingFields(obj2, obj.name, obj.age, obj.job)
```

And reflect the fields to ignore the same field

**Comment by LeoColman** [comment]
Using strings hides the code behind it, and is subject to breaking when one renames the field

**Comment by sksamuel** [comment]
If you can get a function reference by just doing obj.name then yes that's better.

**Comment by LeoColman** [comment]
I'm almost sure we can

**Comment by AlexBroadbent** [comment]
Thanks, excited to start using it!

**Comment by sksamuel** [comment]
Closed via #715
